### DISCUSSION TITLE
How do run an animation on a trigger?

### DISCUSSION THREAD

**Discussion by FriederHannenheim** [slint-ui/slint#4042] [Q&A]
I have a password input field which I want to shake when validation fails. How do I do that? I don't want to change the translation I just want to wiggle it so the user gets some feedback

**Comment by ogoffart** [comment]
This is what I suggest: [Slintpad](https://slintpad.com/?snippet=import+%7B+AboutSlint%2C+Button%2C+VerticalBox+%2C+LineEdit%7D+from+%22std-widgets.slint%22%3B%0Aexport+component+Demo+%7B%0A++++property+%3Cfloat%3E+shaking+%3A+0%3B%0A++++animate+shaking+%7B+duration%3A+0.5s%3B+%7D%0A++++VerticalBox+%7B%0A++++++++alignment%3A+start%3B%0A++++++++HorizontalLayout+%7B+alignment%3A+center%3B+Button+%7B+%0A++++++++++++text%3A+%22OK%21%22%3B+%0A++++++++++++clicked+%3D%3E+%7B%0A++++++++++++++++shaking+%3D+shaking+%3C+0.5+%3F+1+%3A+0%3B%0A++++++++++++%7D%0A++++++++%7D+%7D%0A++++++++Rectangle+%7B%0A++++++++++++height%3A+le.min-height%3B%0A++++++++++++le+%3A%3D+LineEdit+%7B%0A++++++++++++++++x%3A+sin%28shaking*5turn%29+*+1rem%3B%0A++++++++++++++++height%3A+100%25%3B%0A++++++++++++++++width%3A+100%25%3B%0A++++++++++++%7D%0A%0A++++++++%7D%0A++++%7D%0A+++%0A%7D%0A&style=native)
Basically have an animated property and move the position of your widget depending on the value, such that when the value is at ts maximum or minimum it is the same.  (hence the use of `sin`)

There are other ways to achieve that. also with animtaion-ticks.
